injector/dns_disruption.go

```diff
@@ -144,13 +144,8 @@ func (i *DNSDisruptionInjector) Inject() error {
 
 	if i.config.Level == chaostypes.DisruptionLevelPod {
 		if !i.config.OnInit {
-			// write classid to container net_cls cgroup - for iptable filtering
-			if err := i.config.Cgroup.Write("net_cls", "net_cls.classid", chaostypes.InjectorCgroupClassID); err != nil {
-				return fmt.Errorf("error writing classid to pod net_cls cgroup: %w", err)
-			}
-
 			// Redirect traffic marked by targeted InjectorDNSCgroupClassID to CHAOS-DNS
-			if err := i.config.Iptables.AddCgroupFilterRule("OUTPUT", chaostypes.InjectorCgroupClassID, "udp", "53", "CHAOS-DNS"); err != nil {
+			if err := i.config.Iptables.AddCgroupFilterRule("nat", "OUTPUT", i.config.Cgroup.RelativePath(""), "-p", "udp", "--dport", "53", "-j", "CHAOS-DNS"); err != nil {
```

Comment:
Could i check why we no longer need to write `InjectorCgroupClassID` please? 
As per a designated comment, it seems important for the disruption to work.

```
	// InjectorCgroupClassID is linked to the TC tree in the injector network disruption.
	// Also used in the DNS Disruption to allow combined Network + DNS Disruption
	// This value should NEVER be changed without changing the Network Disruption TC tree.
	InjectorCgroupClassID = "0x00020002"
```

Reply:
Another question, what is the effect of adding this cgroup (relative path) into the IpTable Filter?

Consider to extract a method with a method name that explains the effect of adding these rules to help readers 🤗.

Reply:
Joris was a little terse in the PR description:

> Network disruptions don't rely on the cgroup tc filter anymore
Instead, packets are marked by iptables depending on their source cgroup and filtered later by tc with a fw filter classifying packets based on their mark

This approach won't work at all in cgroups v2, so we're completely changing how we mark the packets

Reply:
To complete what Philip said here, it's almost the same mechanism as what we used to do but instead of marking packets through the net_cls cgroup, we're marking packets through iptables depending on the cgroup path of the process creating the packet. And the filter user in tc to use that mark is different but the result is the same.